### PR TITLE
E2E: Add update button click, and more assertions to `can add custom product attributes` test

### LINKED ISSUE #39119: E2E: Add more relevant assertions to the `can add custom product attributes` test
For context: p1688630509029399-slack-C01BWDDTGKX

We should add more assertions at the end of the `can add custom product attributes` test to prevent more issues like https://github.com/woocommerce/woocommerce/issues/39105 from escaping the e2e tests.

### PR DESCRIPTION
**Pull request by rodelgc**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Closes #39119.

This pull request:
- Adds [these assertions](https://github.com/woocommerce/woocommerce/pull/39139/files#diff-43fbdc7512061c5149014b364f067055990b2b8b29083dc42facf15de24f6a79R154-R164) to check for correctness of the added attributes
- Adds [this step](https://github.com/woocommerce/woocommerce/pull/39139/files#diff-43fbdc7512061c5149014b364f067055990b2b8b29083dc42facf15de24f6a79R122-R135) to click on the "Update" button in the Edit Product page.
    - Clicking "Update" will introduce flakiness to succeeding steps because it triggers a lot of requests. To eliminate the flakiness, I added [this line](https://github.com/woocommerce/woocommerce/pull/39139/files#diff-43fbdc7512061c5149014b364f067055990b2b8b29083dc42facf15de24f6a79R134) to wait for the final request to be resolved before proceeding further. Playwright discourages using `networkidle` so I went with this approach instead.

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. I reorganized most of the steps in `create-product-attributes.spec.js` so please review if this new flow makes sense.
1. Checkout this branch.
2. Run our usual commands for launching the local E2E environment.
3. Run the `create-product-attributes.spec.js` spec and make sure it's not flaky by running multiple instances of it repeatedly. You can do so through this command:
    ```bash
    cd plugins/woocommerce
    pnpm test:e2e-pw create-product-attributes.spec.js --retries=0 --workers=4 --repeat-each=8
    ```
4. All instances of this test should pass, which indicates it isn't flaky.
1. Run this spec together with the rest of the `merchant` tests:
    ```bash
    cd plugins/woocommerce
    pnpm test:e2e-pw merchant
    ```
    They should all pass.

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce/tests/e2e-pw/tests/merchant/products/add-variable-product/create-product-attributes.spec.js [hunk 1]
```diff
@@ -23,109 +37,130 @@ test.describe( 'Add product attributes', () => {
...
+				await test.step( "Click 'Add new'.", async () => {
+					await page
+						.getByRole( 'button', { name: 'Add new' } )
+						.click();
+
+					await page
+						.getByRole( 'heading', {
+							name: 'New attribute',
+						} )
+						.waitFor();
```
Comment (lanej0) [inline plugins/woocommerce/tests/e2e-pw/tests/merchant/products/add-variable-product/create-product-attributes.spec.js]: Are we waiting for this to be visible?  Might be better to be explicit about it: https://playwright.dev/docs/testing-library#replacing-waitfor

**Review by lanej0 — APPROVED** [review]
Looks good to me.  I had one non-blocking question, so I won't merge just yet but this is approved!

File: plugins/woocommerce/tests/e2e-pw/tests/merchant/products/add-variable-product/create-product-attributes.spec.js (hunk 1 above)
Comment (rodelgc) [reply plugins/woocommerce/tests/e2e-pw/tests/merchant/products/add-variable-product/create-product-attributes.spec.js]: Oh, thank you very much! Yeah it's much better to be more explicit with the visibility check. Done on 3d923b13e7df08ba2c6b1bbd2a47b0b7f4e2d77a. And thanks, I learned something new 😄

**Review by lanej0 — APPROVED** [review]
Looks good!

**Pull request merged**